addons/uom/models/uom_uom.py

```diff
@@ -23,7 +23,7 @@ def _onchange_uom_ids(self):
             self.uom_ids[0].uom_type = 'reference'
             self.uom_ids[0].factor = 1
         else:
-            reference_count = sum(uom.uom_type == 'reference' for uom in self.uom_ids)
+            reference_count = sum(uom.uom_type == 'reference' for uom in self.uom_ids or self._origin.uom_ids)
```

Comment:
```suggestion
            reference_count = sum(uom.uom_type == 'reference' for uom in (self.uom_ids or self._origin.uom_ids))
```

Reply:
@MiquelRForgeFlow Thanks, but i already test in command line, i think we don't actually need the ()

Reply:
```suggestion
            reference_count = sum(uom.uom_type == 'reference' for (uom in self.uom_ids) or (uom in self._origin.uom_ids))
```